CHANGELOG.md

```diff
@@ -1,3 +1,1038 @@
+directive attributes).
+
+- **feat($q): report promises with non rejection callback ([c9dffd](https://github.com/angular/angular.js/commit/c9dffde1cb167660120753181cb6d01dc1d1b3d0))**:
+
+Unhandled rejected promises will be logged to $exceptionHandler.
+
+Tests that depend on specific order or number of messages in $exceptionHandler
+will need to handle rejected promises report.
+
+Closes: #13653
+Closes: #7992
+
+- **feat($location): default hashPrefix to '!' ([aa077e](https://github.com/angular/angular.js/commit/aa077e81129c740041438688dff2e8d20c3d7b52))**:
+
+The hash-prefix for `$location` hash-bang URLs has changed from the empty
+string &quot;&quot; to the bang &quot;!&quot;. If your application does not use HTML5 mode
+or is being run on browsers that do not support HTML5 mode, and you have
+not specified your own hash-prefix then client side URLs will now contain
+a &quot;!&quot; prefix. For example, rather than `mydomain.com/#/a/b/c` will become
+`mydomain/#!/a/b/c`.
+
+If you actually wanted to have no hash-prefix then you should configure
+this by adding a configuration block to you application:
+
+```
+appModule.config(['$locationProvider', function($locationProvider) {
+  $locationProvider.hashPrefix(&quot;&quot;);
+}]);
+```
+- **fix(select): don't register options when select has no ngModel ([e8c2e1](https://github.com/angular/angular.js/commit/e8c2e119758e58e18fe43932d09a8ff9f506aa9d))**:
+
+Option elements will no longer set their value attribute from their text value when their select
+element has no ngModel associated. Setting the value is only needed for the select directive to
+match model values and options. If no ngModel is present, the select directive doesn't need it.
+
+This should not affect many applications as the behavior was undocumented and not part of a public
+API. It also has no effect on the usual HTML5 behavior that sets the select value to the option text
+if the option does not provide a value attribute.
+
+- **fix($resource): add semicolon to whitelist of delimiters to unencode in URL params ([2456ab](https://github.com/angular/angular.js/commit/2456ab63a613902d21c151445f9c697a76ab43b3))**:
+
+Although it shouldn't matter in practice (since both the encoded and the unencoded `;` character would
+be interpreted identically by the server), this change could break some tests: For example, where
+`$httpBackend` was set up to expect an encoded `;` character, but the request is made to the URL with an
+unencoded `;` character.
+
+- **feat(input): add support for binding to `input[type=range]` ([913016](https://github.com/angular/angular.js/commit/9130166767c4792c5d32d08a918fc7becf32c9a6))**:
+
+Due to the way that `input[type=range]` elements behave this feature modifies the behavior of such elements
+when bound to `ngModel`:
+
+- Like input[number], it requires the model to be a Number, and will set the model to a Number
```

Comment:
_input[number]_ --> _\`input[number]\`_ (or else `[number]` is interpreted as link)